user/src/main/java/bisq/user/identity/UserIdentityStore.java

```diff
@@ -44,7 +44,9 @@ public UserIdentityStore() {
 
     private UserIdentityStore(UserIdentity selectedUserIdentity,
                               Set<UserIdentity> userIdentities) {
-        this.selectedUserIdentity.set(selectedUserIdentity);
+        this.selectedUserIdentity.set(userIdentities.stream()
```

Comment:
We should find the reason why a different instance of (the same) userIdentity is used. I guess we might get it from another persisted object (channelSelection?).

Reply:
When userIdentityStore is loaded from file it creates selectedUserIdentity from protobuf and the userIdentities Set from protobuf separately. I think this is ok, as long as the selectedUserIdentity instance is only used to select the instance from the Set.

I haven't seen any other usages where selectedUserIdentity would be set to an instance not in the Set, but it would be possible. I added a setter to make sure an instance from the Set is used.

Reply:
Ah ok. We should replace the selectedUserIdentity with an selectedUserIdentityId (pubkeyHash) and use that to lookup the one from the set. We have a similar pattern used for channels.

Reply:
But this can be done a new PR...

Reply:
I don't think it's necessary to use the hash. Saving the object makes it easier to see what is saved if you look at the save files and it needs to be handled when loaded in any case.